### PR TITLE
Homework generate sequence Lukasz K/Nanautzin

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: generate-sequence
🏅 Granted 5 XP for @lukasz-kukulka

**Comment by ziobron** [comment]
⏱ 3 XP extra for @lukasz-kukulka for on-time delivery

File: homework/generate-sequence/generateSequence.hpp [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 #include <vector>
+#include <algorithm>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    if (count <= 0){
+        std::vector<int> vec;
+        return vec;
+    } else {
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:9]: If you return variable here, you don't need to use `else if` or `else`

You can return {}. Compiler will automatically construct empty vector.

File: homework/generate-sequence/generateSequence.hpp [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+    std::vector<int> vec(count);
+    auto results = 0;
+    std::generate(vec.begin(), vec.end(), [&step, &results]() { 
+        results += step;
+        return results;
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:14]: Nice!!!

**Review by nauka-programowania-MA — APPROVED** [review]
Perfect! I really enjoy your solutions!

File: homework/generate-sequence/generateSequence.hpp (hunk 1 above)
Comment (lukasz-kukulka) [reply homework/generate-sequence/generateSequence.hpp:9]: > If you return variable here, you don't need to use `else if` or `else`
> 
> You can return {}. Compiler will automatically construct empty vector.

Thank you for sugestion

**Pull request closed without merging**